engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,269 @@
+     * @return a new FilterSelectColumn representing the provided filter.
+     */
+    static FilterSelectColumn of(@NotNull final String destName, @NotNull final WhereFilter filter) {
+        return new FilterSelectColumn(destName, filter);
+    }
+
+    private FilterSelectColumn(@NotNull final String destName, @NotNull final WhereFilter filter) {
+        this.destName = destName;
+        this.filter = filter;
+    }
+
+    @Override
+    public String toString() {
+        return "filter(" + filter + ')';
+    }
+
+    @Override
+    public List<String> initInputs(TrackingRowSet rowSet, Map<String, ? extends ColumnSource<?>> columnsOfInterest) {
+        this.rowSet = rowSet;
+        this.tableToFilter = new QueryTable(rowSet, columnsOfInterest);
+        return filter.getColumns();
+    }
+
+    @Override
+    public List<String> initDef(@NotNull Map<String, ColumnDefinition<?>> columnDefinitionMap) {
+        filter.init(TableDefinition.of(columnDefinitionMap.values()));
+        return checkForInvalidFilters();
+    }
+
+    @Override
+    public List<String> initDef(@NotNull final Map<String, ColumnDefinition<?>> columnDefinitionMap,
+            @NotNull final QueryCompilerRequestProcessor compilationRequestProcessor) {
+        filter.init(TableDefinition.of(columnDefinitionMap.values()), compilationRequestProcessor);
+        return checkForInvalidFilters();
+    }
+
+    /**
+     * Validates the filter to ensure it does not contain invalid filters such as column vectors or virtual row
+     * variables. Throws an {@link UncheckedTableException} if any invalid filters are found.
+     *
+     * @return the list of columns required by the filter.
+     */
+    private List<String> checkForInvalidFilters() {
+        if (!filter.getColumnArrays().isEmpty()) {
+            throw new UncheckedTableException(
+                    "Cannot use a filter with column Vectors (_ syntax) in select, view, update, or updateView: "
+                            + filter);
+        }
+        if (filter.hasVirtualRowVariables()) {
+            throw new UncheckedTableException(
+                    "Cannot use a filter with virtual row variables (i, ii, or k) in select, view, update, or updateView: "
+                            + filter);
+        }
+
+        return filter.getColumns();
+    }
+
+    @Override
+    public Class<?> getReturnedType() {
+        return Boolean.class;
+    }
+
+    @Override
+    public Class<?> getReturnedComponentType() {
+        return null;
+    }
+
+    @Override
+    public List<String> getColumns() {
+        return filter.getColumns();
+    }
+
+    @Override
+    public List<String> getColumnArrays() {
+        return List.of();
+    }
+
+    @Override
+    public boolean hasVirtualRowVariables() {
+        /* This should always be false, because initDef throws when arrays or ii and friends are used. */
+        return filter.hasVirtualRowVariables();
```

Comment:
So, I think we should be consistent in how we implement these two methods; either delegate both with a comment, or return empty list and false literal (can still have a comment with literals if you want).